src/Microsoft.ML.PCA/PcaTransform.cs

```diff
@@ -493,56 +530,134 @@ private Float[][] PostProcess(Float[][] y, Float[] sigma, Float[] z, int d, int
             return y;
         }
 
-        private ColumnType[] InitColumnTypes()
+        protected override IRowMapper MakeRowMapper(ISchema schema) => new Mapper(this, Schema.Create(schema));
+
+        protected override void CheckInputColumn(ISchema inputSchema, int col, int srcCol)
         {
-            Host.Assert(Infos.Length == _transformInfos.Length);
-            var types = new ColumnType[Infos.Length];
-            for (int i = 0; i < _transformInfos.Length; i++)
-                types[i] = new VectorType(NumberType.Float, _transformInfos[i].Rank);
-            Metadata.Seal();
-            return types;
+            ValidatePcaInput(Host, inputSchema.GetColumnName(srcCol), inputSchema.GetColumnType(srcCol));
         }
 
-        protected override ColumnType GetColumnTypeCore(int iinfo)
+        internal static void ValidatePcaInput(IHost host, string name, ColumnType type)
         {
-            Host.Check(0 <= iinfo & iinfo < Utils.Size(_types));
-            return _types[iinfo];
+            if (!type.IsVector)
+                throw host.Except($"Pca transform can only be applied to vector columns. Column ${name} is of type ${type}");
```

Comment:
>host.Except( [](start = 22, length = 12)

I suggest to refactor this slightly to fit into `ExceptSchemaMismatch`

Reply:
There wasn't a good type to to put for the expected type. The best I could come up with was this, which didn't look good.

            string inputSchema; // just used for the excpections

            if (!type.IsVector)
                throw ectx.ExceptSchemaMismatch(nameof(inputSchema), "input", name, "Vector", type.ToString());

---
In reply to: [227587808](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227587808) [](ancestors = 227587808)

Reply:
Well, you are expecting a fixed-size vector of float. You can either spell it out 'fixed size vector of float', or create a `SchemaShape.Column` and call `GetTypeString`

---
In reply to: [227630278](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227630278) [](ancestors = 227630278,227587808)

Reply:
Consolidated all the checks.

---
In reply to: [227877917](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227877917) [](ancestors = 227877917,227630278,227587808)